lib/presentation/features/home/bloc/home_bloc.dart

```diff
@@ -1,28 +1,47 @@
+import 'package:flutter/widgets.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:marvelapp_flutter/domain/use_cases/get_characters_use_case.dart';
 import 'package:marvelapp_flutter/presentation/features/home/bloc/home_event.dart';
 import 'package:marvelapp_flutter/presentation/features/home/bloc/home_state.dart';
 import 'package:marvelapp_flutter/presentation/extensions/to_character_view_data.dart';
+import 'package:marvelapp_flutter/presentation/models/character_view_data.dart';
 
 class HomeBloc extends Bloc<HomeEvent, HomeState> {
   final GetCharactersUseCase getCharactersUseCase;
 
-  HomeBloc({required this.getCharactersUseCase}) : super(const HomeState(loading: true)) {
+  HomeBloc({required this.getCharactersUseCase}) : super(const HomeState(loading: false)) {
     on<GetHeroes>(_mapGetHeroesEventToState);
   }
 
   Future<void> _mapGetHeroesEventToState(GetHeroes event, Emitter<HomeState> emit) async {
-    emit(state.copyWith(loading: true, error: null));
+    if (state.hasReachedMax) return;
+    if (state.loading) return;
+    if (state.characters == null) {
+      emit(state.copyWith(loading: true));
+    } else {
+      emit(state.copyWith(loading: true, error: null));
+    }
```

Comment:
This looks unclear, could you eleborate on it? What does it mean?
```
    if (state.characters == null) {
      emit(state.copyWith(loading: true));
    } else {
      emit(state.copyWith(loading: true, error: null));
    }
```

Reply:
remove it and change to one emit. It apeared because at first time i dont understand how does it work. and i try control it all process in view and in bloc.

Reply:
done